eras/mary/impl/src/Cardano/Ledger/Mary/BlockBody.hs

```diff
@@ -0,0 +1,27 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Cardano.Ledger.Mary.BlockBody () where
+
+import Cardano.Ledger.Core (EraBlockBody (..))
+import Cardano.Ledger.Mary.Era (MaryEra)
+import Cardano.Ledger.Mary.Tx ()
+import Cardano.Ledger.Shelley.BlockBody.Internal (
+  ShelleyBlockBody (ShelleyBlockBody),
+  sbbHash,
+  sbbTxs,
+ )
```

Comment:
Instead of importing an `Internal` module, you need to create and export safe versions of these functions.
```suggestion
import Cardano.Ledger.Shelley.BlockBody (
  ShelleyBlockBody (ShelleyBlockBody),
  shelleyBlockBodyHash,
  shelleyBlockBodyTxs,
 )
```

This is a general rule of thumb. If we don't have to rely on the internals, we should do our best to use safe functionality. Not only it promotes safer code, but it also serves as an example for any downstream user that is reading our code.